libr/core/cmd_eval.c

```diff
@@ -102,32 +102,35 @@ static bool nextpal_item(RCore *core, int mode, const char *file, int ctr) {
 	return true;
 }
 
-R_API RList *r_core_list_themes(RCore *core) {
-	RList *files = NULL;
+static void list_themes_in_path(RList *list, const char *path) {
 	RListIter *iter;
 	const char *fn;
-	char *home = r_str_home (".config/radare2/cons/");
-
-	RList *list = r_list_new ();
-	getNext = false;
-	if (home) {
-		files = r_sys_dir (home);
-		r_list_foreach (files, iter, fn) {
-			if (*fn && *fn != '.') {
-				r_list_append (list, strdup (fn));
-			}
-		}
-		r_list_free (files);
-		R_FREE (home);
-	}
-	files = r_sys_dir (R2_DATDIR"/radare2/"R2_VERSION"/cons/");
+	RList *files = r_sys_dir (path);
 	r_list_foreach (files, iter, fn) {
 		if (*fn && *fn != '.') {
 			r_list_append (list, strdup (fn));
 		}
 	}
 	r_list_free (files);
-	files = NULL;
+	R_FREE (path);
+}
+
+R_API RList *r_core_list_themes(RCore *core) {
+	RList *list = r_list_new ();
```

Comment:
use r_list_newf (free), so we ensure that all strings are freed when the list is freed